### PR TITLE
add action prop to Button

### PR DESCRIPTION
**Pull request by souporserious**
**What**:
This adds an `action` prop to the `Button` component.

**Why**:
After messing with the example I realized `clearSelection` didn't work and figured we should bake this in since there is some proper coordination that needs to go on with key handlers and click events. This also widens the pit of success for accessibility 🎉 

**How**:
I added a map of how each action should be handled and then call the respective action with the prop that was passed. I think after this we can explore even better accessibility and expose the labels for internationalization 😁 

**Checklist**:

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged 
- [ ] Added myself to contributors table

### CODE REVIEW

**Comment by kentcdodds** [comment]
Thanks for this! After considering it I would like to come up with a different approach. I really don't like strings that have any significant meaning in a codebase. I think it makes for a fragile API. So instead what if you could do:

```js
action={Autocomplete.Button.open}
```

That would allow you to do:

```js
action={() => {
  Autocomplete.Button.open()
  console.log('opened')
}}
```

And in that case, what if we switch `action` with `onClick`:

```js
onClick={Autocomplete.Button.open}
```

And then what if we just expose the actions in the controller?

```javascript
<Autocomplete.Controller>
{({open}) => (
  <Autocomplete.Button onClick={open}>Blah</Autocomplete.Button>
)}
</Autocomplete.Controller>
```

Oh! I think that this already exists! :wink:

What do you think?

**Comment by souporserious** [comment]
Totally. I was trying to figure out if there was an easier way to promote accessibility. Maybe just require an `aria-label`?

**Comment by kentcdodds** [comment]
Yeah, that's probably best. Just have a prop-types for it maybe?

File: src/button.js [hunk 1]
```diff
@@ -38,11 +59,13 @@ class Button extends Component {
     },
 
     Enter(event) {
-      event.preventDefault()
-      if (this.autocomplete.state.isOpen) {
-        this.autocomplete.selectHighlightedItem()
-      } else {
-        this.autocomplete.openMenu()
+      if (this.props.action !== 'clearSelection') {
```
Comment (souporserious) [inline src/button.js]: Any ideas how to handle this condition if we aren't using an action prop?

Comment (souporserious) [inline src/button.js]: @kentcdodds any ideas how to handle this condition if we aren't using an action prop?

**Comment by souporserious** [comment]
This seems too opinionated and the API is weird like you said so I'm closing this.

**Pull request closed without merging**